packages/expo-web-browser/src/WebBrowser.ts

```diff
@@ -33,7 +42,44 @@ export async function getCustomTabsSupportingBrowsersAsync(): Promise<CustomTabs
   if (!ExponentWebBrowser.getCustomTabsSupportingBrowsersAsync) {
     throw new UnavailabilityError('WebBrowser', 'getCustomTabsSupportingBrowsersAsync');
   }
-  return ExponentWebBrowser.getCustomTabsSupportingBrowsersAsync();
+  if (Platform.OS === 'android') {
+    return ExponentWebBrowser.getCustomTabsSupportingBrowsersAsync();
+  } else {
+    return emptyCustomTabsPackages;
+  }
+}
+
+export async function warmUpAsync(browserPackage?: string) {
+  if (!ExponentWebBrowser.warmUpAsync) {
+    throw new UnavailabilityError('WebBrowser', 'warmUpAsync');
+  }
+  if (Platform.OS === 'android') {
+    return ExponentWebBrowser.warmUpAsync(browserPackage);
+  } else {
+    return null;
+  }
+}
+
+export async function mayInitWithUrlAsync(url: string, browserPackage?: string) {
+  if (!ExponentWebBrowser.mayInitWithUrlAsync) {
+    throw new UnavailabilityError('WebBrowser', 'mayInitWithUrlAsync');
+  }
+  if (Platform.OS === 'android') {
+    return ExponentWebBrowser.mayInitWithUrlAsync(url, browserPackage);
+  } else {
+    return null;
+  }
+}
+
+export async function coolDownAsync(browserPackage?: string) {
+  if (!ExponentWebBrowser.coolDownAsync) {
+    throw new UnavailabilityError('WebBrowser', 'coolDownAsync');
+  }
+  if (Platform.OS === 'android') {
```

Comment:
Could you also please change the logic of these `if`s to how it is designed in `expo-brightness` which is after the audit and is supposed to be our good-quality-module example (https://github.com/expo/expo/pull/2408)?

https://github.com/expo/expo/blob/master/packages/expo-brightness/src/Brightness.ts#L47-L52